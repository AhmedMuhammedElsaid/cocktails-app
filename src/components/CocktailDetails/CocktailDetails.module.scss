@import "../../scss//variables.scss";

.CocktailDetails {
  &__Ingredient {
    cursor: pointer;
    color: $darkGrey;
    text-decoration: underline;
    font-weight: 700;
    &:hover {
      color: $mainOrange;
    }
  }
  &__Info {
    font-size: 1.3rem;
    line-height: 1.85;
    color: $darkGrey;

    span {
      opacity: 0.8;
      color: $mainBlack;
      font-weight: 800;
    }
  }
}
