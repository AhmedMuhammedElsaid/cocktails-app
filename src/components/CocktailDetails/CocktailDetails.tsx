import React, { FunctionComponent } from "react";
import { Typography } from "@material-ui/core";
import styles from "./CocktailDetails.module.scss";
import { cocktailSchema } from "utils/types";

// component to render the cocktail details
interface Props {
  currentCocktail: cocktailSchema;
  handleGetSimilarCocktailToIngredient: (value: string) => void;
}
const CocktailDetails: FunctionComponent<Props> = ({
  currentCocktail,
  handleGetSimilarCocktailToIngredient,
}) => {
  const grabImportantDetails: any = { ...currentCocktail };
  /*
  we loop over the cocktail props but we dont need to repeat [name,category , image] 
  as we already rendered them on the top
  */
  delete grabImportantDetails?.strDrinkThumb;

  delete grabImportantDetails?.strDrink;
  delete grabImportantDetails?.strImageSource;
  delete grabImportantDetails?.strCategory;
  return (
    <div data-testid="cocktail-details-testid">
      {Object.keys(grabImportantDetails).map(
        (property: string, index: number) => {
          if (property.includes("strIngredient")) {
            // in case the property is ingredient , i want it to be clickable to be able to see similar cocktails which use that ingredient
            return (
              <Typography
                key={index}
                onClick={() =>
                  handleGetSimilarCocktailToIngredient(
                    grabImportantDetails[property]
                  )
                }
                className={styles.CocktailDetails__Ingredient}
                variant="h6"
                // removing str from the beginning of every property
              >{`${property?.slice(3, property.length)} : ${
                grabImportantDetails[property]
              }`}</Typography>
            );
          } else {
            // otherwise i want just to render it
            return (
              <Typography
                className={styles.CocktailDetails__Info}
                key={index}
                variant="h6"
              >
                <span>{`${
                  // removing str from the beginning of every property
                  property.includes("str")
                    ? property.slice(3, property.length)
                    : property
                } : `}</span>
                {`${grabImportantDetails[property]?.slice(0, 150)}`}
              </Typography>
            );
          }
        }
      )}
    </div>
  );
};
export default CocktailDetails;
