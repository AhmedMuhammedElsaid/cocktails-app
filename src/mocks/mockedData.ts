export const cocktailDetailsMockedData = {
  idDrink: "14608",
  strDrink: "Cream Soda",
  strCategory: "Cocktail",
  strAlcoholic: "Alcoholic",
  strGlass: "Highball glass",
  strInstructions:
    "Pour 1oz of Spiced Rum into a highball glass with ice. Fill with Ginger Ale.",
  strInstructionsDE:
    "Geben Sie 3 cl. Spiced Rum in ein Highball-Glas mit Eis. Mit Ginger Ale .",
  strInstructionsIT: `Versare 30gr di rum speziato in un bicchiere highball con ghiaccio.Riempi con Ginger Ale.`,
  strDrinkThumb: `https://www.thecocktaildb.com/images/media/drink/yqstxr1479209367.jpg`,
  strIngredient1: `Spiced rum`,
  strIngredient2: `Ginger ale`,
  strCreativeCommonsConfirmed: `No`,
  dateModified: `2016-11-15 11:29:27`,
};
export const cocktailListMockedData = [
  cocktailDetailsMockedData,
  { ...cocktailDetailsMockedData, idDrink: "14709" },
];
export const cocktailCardMockedData = {
  strDrink: "Cream Soda",
  strCategory: "Cocktail",
  strDrinkThumb: `https://www.thecocktaildb.com/images/media/drink/yqstxr1479209367.jpg`,
  strInstructions:
    "Geben Sie 3 cl. Spiced Rum in ein Highball-Glas mit Eis. Mit Ginger Ale .",
};
