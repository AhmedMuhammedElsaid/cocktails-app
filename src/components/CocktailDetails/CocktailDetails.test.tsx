import { render } from "@testing-library/react";
import { cocktailDetailsMockedData } from "mocks/mockedData";
import CocktailDetails from "./CocktailDetails";

describe("test cocktail details component", () => {
  it("render cocktail details   without error", () => {
    const { getByTestId,asFragment } = render(<CocktailDetails handleGetSimilarCocktailToIngredient={jest.fn()} currentCocktail={cocktailDetailsMockedData} />);
    expect(asFragment()).toMatchSnapshot()
  });
  it("render cocktail details without passing data", () => {
    const { getByTestId,asFragment } = render(<CocktailDetails handleGetSimilarCocktailToIngredient={jest.fn()}  />);
    expect(asFragment()).toMatchSnapshot()
  });
});
