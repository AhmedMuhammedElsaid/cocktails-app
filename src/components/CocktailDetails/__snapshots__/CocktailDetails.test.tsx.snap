// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cocktail details component render cocktail details   without error 1`] = `
<DocumentFragment>
  <div
    data-testid="cocktail-details-testid"
  >
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        idDrink : 
      </span>
      14608
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        Alcoholic : 
      </span>
      Alcoholic
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        Glass : 
      </span>
      Highball glass
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        Instructions : 
      </span>
      Pour 1oz of Spiced Rum into a highball glass with ice. Fill with Ginger Ale.
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        InstructionsDE : 
      </span>
      Geben Sie 3 cl. Spiced Rum in ein Highball-Glas mit Eis. Mit Ginger Ale .
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        InstructionsIT : 
      </span>
      Versare 30gr di rum speziato in un bicchiere highball con ghiaccio.Riempi con Ginger Ale.
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Ingredient MuiTypography-h6"
    >
      Ingredient1 : Spiced rum
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Ingredient MuiTypography-h6"
    >
      Ingredient2 : Ginger ale
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        CreativeCommonsConfirmed : 
      </span>
      No
    </h6>
    <h6
      class="MuiTypography-root CocktailDetails__Info MuiTypography-h6"
    >
      <span>
        dateModified : 
      </span>
      2016-11-15 11:29:27
    </h6>
  </div>
</DocumentFragment>
`;

exports[`test cocktail details component render cocktail details without passing data 1`] = `
<DocumentFragment>
  <div
    data-testid="cocktail-details-testid"
  />
</DocumentFragment>
`;
